card/mod/05_email/set/all/follow.rb

```diff
@@ -167,13 +179,15 @@ def direct_followers
 
 # all ids of users that follow this card because of a follow rule that applies to this card
 # doesn't include users that follow this card because they are following parent cards or other cards that include this card
-def direct_follower_ids args={}  
+def direct_follower_ids args={}
   result = ::Set.new
-  set_names.each do |set_name| 
-    set_card = Card.fetch(set_name)
-    set_card.all_user_ids_with_rule_for(:follow).each do |user_id|
-      if (!result.include? user_id) and self.follow_rule_applies?(user_id)
-        result << user_id
+  set_follow_test_options do
```

Comment:
this was very confusing to me. "set" appears six times in various names in this method but in all other cases it means card set.